### ISSUE TITLE
Code completion from Julia Kernel

### ISSUE THREAD

**Issue by DonJayamanne** [julia-vscode/julia-vscode#2051]
I understand theres a language server, however based on my experience, the completions are always based on static analysis.
The Jupyter protocol supports getting completions from the kernels.

Is this something we want to support? Hooking these up in the extension is fairly easy, but we'd need some code in the julia kernel to handle these requests, if not already supported.

The assumption is that the code completion we (can) get from the kernel will be better than that we get from the language server. (again this is based on my assumptions about the language semantics and how the existing language server works).

**Comment by davidanthoff** [comment]
Yeah, we've been talking about this on and off over the years, that we could augment the code completion we get from the static analysis in the LS with dynamic info from the Julia REPL or now a notebook kernel. I think my sense is that if we were to implement that we would do it in one go for the REPL and the notebooks. My vote would be that we leave this on the backlog, and then just see whether it fits in the general LS dev story we have going on. Right now we have some major other projects going on with the LS that I think we need to finish before we consider doing something like this. Having said that, I agree that this would probably be a good idea.

**Comment by pfitzseb** [comment]
Ref https://github.com/julia-vscode/julia-vscode/pull/1507.

Getting the completions is easy, dealing with "merge conflicts" isn't :)

**Comment by davidanthoff** [comment]
One other thought on this: I think this kind of dynamic completion is particularly helpful with code in the global context, in function I don't think it adds a lot. But of course in notebooks one very often has a lot of global code, so I think in that context this would be especially helpful.

**Comment by FabrizioTorrico** [comment]
What were your conclusions after these years?
